js/interface.js

```diff
@@ -554,12 +554,26 @@ $('#app')
   .on('click', '[data-back]', function(event) {
     event.preventDefault();
     $('[href="#entries"]').click();
-    if (!dataSourceEntriesHasChanged || confirm('Are you sure? Changes that you made may not be saved.')) {
-      try{
-        table.destroy();
-      } catch(e) {
-      }
-      dataSourceEntriesHasChanged = false;
+
+    if (dataSourceEntriesHasChanged) {
+      Fliplet.Modal.confirm({
+        message: 'Are you sure? Changes that you made may not be saved.'
+      }).then(function(result) {
+        if (!result) {
+          return;
+        }
+
+        try {
+          table.destroy();
+        } catch(e) {}
+
+        dataSourceEntriesHasChanged = false;
+        $('.data-save-updated').addClass('hidden');
```

Comment:
why were these 3 lines below necessary?

Reply:
Those lines navigate user to all data sources. I had to duplicate them to make both flows (saved, and not saved) to work properly.